### PR TITLE
bootstrap: disallow upgrade from below 6.2 to current master

### LINKED ISSUE #56658: disable upgrade from below 6.2 to current master（8.5+）
## Enhancement
since v6.2.0 and bootstrap version 92, we start support [concurrent DDL](https://github.com/pingcap/tidb/issues/32031) where
DDL jobs are stored in system table instead of plain meta KV queue. we have
kept the old code for 4 LTS versions to make upgrade smooth as much as possible.
Since v8.5, we will remove those old code. If you want to upgrade from below
v6.2.0 to version >= v8.5, you need to upgrade to a version in range [v6.2.0, v8.5.0)
first, then upgrade to v8.5+.

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: close #56658

Problem Summary:

### What changed and how does it work?
as title. code related to old ddl job queue will removed in later PR
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

upgrade from 6.1.0 -> master, fail with error:
```[2024/10/15 18:32:46.726 +08:00] [FATAL] [terror.go:309] ["unexpected error"] [error="Cannot upgrade from below v6.2.0 to this version, please upgrade to a version in range [v6.2.0, v8.5.0) first. Current bootstrap version is 91, target version is 216"] [stack="............."]```

upgrade from 6.2.0/8.1.0 -> master, success

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/session/session.go [hunk 1]
```diff
@@ -3836,7 +3846,7 @@ func getStoreBootstrapVersion(store kv.Storage) int64 {
 	ctx := kv.WithInternalSourceType(context.Background(), kv.InternalTxnBootstrap)
 	err := kv.RunInNewTxn(ctx, store, false, func(_ context.Context, txn kv.Transaction) error {
 		var err error
-		t := meta.NewMutator(txn)
+		t := meta.NewReader(txn)
```
Comment (Benjamin2037) [inline pkg/session/session.go:3849]: Does this line of change related this pr？

Comment (D3Hunter) [reply pkg/session/session.go:3849]: no, just found and change it by the way

Comment (Benjamin2037) [reply pkg/session/session.go:3849]: Got it！

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Comment by Benjamin2037** [comment]
May need let Tester who own upgrade testing know this change and do changes according.

**Review by tangenta — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]

**Comment by D3Hunter** [comment]
just found out that we have already remove the code of `DDLForce2Queue=true` when forcing the new node to be owner in  https://github.com/pingcap/tidb/pull/51285/files#diff-2fde3341e0231e7b162880b36f07d23b7949ca512c5536b50706ac5aa9072b25L1484

so the limitation in this PR is not needed

**Pull request closed without merging**